rdkit/Chem/TorsionFingerprints.py

```diff
@@ -123,7 +139,8 @@ def _getIndexforTorsion(neighbors, inv):
   elif _doMatch(inv, neighbors): # atom has all symmetric neighbors
     return neighbors
   elif _doNotMatch(inv, neighbors): # atom has all different neighbors
-    # simply use the first neighbor
+    # sort by atom inv and simply use the first neighbor
+    neighbors = _sortNeighbors(inv, neighbors)
```

Comment:
You could make this a lot more efficient (and more readable) by taking advantage of the features of Python's sort function. Something like this should work:
```neighbors = sorted(neighbors,key=lambda x:inv[x.GetIdx()])```